### PR TITLE
Add marriage-related tax reforms

### PR DESCRIPTION
**Pull request by nikhilwoodruff**

### CODE REVIEW

File: policyengine_uk/reforms/cps/marriage_tax_reforms.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+        def formula(person, period):
+            band = person("tax_band", period)
+            bands = band.possible_values
+            return (
+                (band == bands.BASIC)
+                | (band == bands.STARTER)
+                | (band == bands.INTERMEDIATE)
+                | person("meets_expanded_ma_conditions", period)
+                & ((band == bands.HIGHER) | (band == bands.ADDITIONAL))
```
Comment (MaxGhenis) [inline policyengine_uk/reforms/cps/marriage_tax_reforms.py]: can you add parens to clarify this behavior? given the | and &

File: policyengine_uk/reforms/cps/marriage_tax_reforms.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+                child_meets_age_condition = (
+                    person("age", period) <= max_child_age
+                )
+                meets_conditions = benunit.any(child_meets_age_condition)
+            if child_education_levels is not None:
+                child_meets_education_condition = np.is_in(
+                    person("education_level", period).decode_to_str(),
+                    child_education_levels,
+                )
+                meets_conditions = benunit.any(child_meets_education_condition)
```
Comment (MaxGhenis) [inline policyengine_uk/reforms/cps/marriage_tax_reforms.py]:
```suggestion
                return benunit.any(child_meets_education_condition)
```

File: policyengine_uk/reforms/cps/marriage_tax_reforms.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            )
+            # Round up.
+            rounding_increment = (
+                allowances.marriage_allowance.rounding_increment
+            )
+            amount_if_eligible = (
+                np.ceil(amount_if_eligible_pre_rounding / rounding_increment)
+                * rounding_increment
+            )
+            return (
```
Comment (MaxGhenis) [inline policyengine_uk/reforms/cps/marriage_tax_reforms.py]: just for clarity
```suggestion
            takes_up = random(person) < takeup_rate
            return eligible * amount_if_eligible * takes_up
```

File: policyengine_uk/reforms/cps/marriage_tax_reforms.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+                child_meets_age_condition = (
+                    person("age", period) <= max_child_age
+                )
+                meets_conditions = benunit.any(child_meets_age_condition)
+            if child_education_levels is not None:
+                child_meets_education_condition = np.is_in(
+                    person("education_level", period).decode_to_str(),
+                    child_education_levels,
+                )
+                meets_conditions = benunit.any(child_meets_education_condition)
```
Comment (MaxGhenis) [inline policyengine_uk/reforms/cps/marriage_tax_reforms.py]:
```suggestion
                return benunit.any(child_meets_education_condition)
```

**Review by MaxGhenis — COMMENTED** [review]
How does this work in terms of only loading the reform when activated?

File: policyengine_uk/reforms/cps/marriage_tax_reforms.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    remove_income_condition = cps.expanded_ma.remove_income_condition
+    rate = cps.expanded_ma.ma_rate
+    original_rate = parameters(
+        period
+    ).gov.hmrc.income_tax.allowances.marriage_allowance.max
+    ma_max_child_age = cps.expanded_ma.max_child_age
+    neutralise_income_tax = cps.marriage_neutral_it.neutralise_income_tax
+    it_max_child_age = cps.marriage_neutral_it.max_child_age
+
+    if remove_income_condition or rate != original_rate:
```
Comment (nikhilwoodruff) [inline policyengine_uk/reforms/cps/marriage_tax_reforms.py:223]: @MaxGhenis here's where we apply the reform if the parameters are switched on

**Pull request merged**